src/metpy/calc/thermo.py

```diff
@@ -1556,14 +1556,66 @@ def saturation_vapor_pressure(temperature):
     Instead of temperature, dewpoint may be used in order to calculate
     the actual (ambient) water vapor (partial) pressure.
 
-    The formula used is that from [Bolton1980]_ for T in degrees Celsius:
+    Implemented solution from [Ambaum2020]_, Eq. 13,
+    .. math:: e = e_{s0} \frac{T_0}{T}^{(c_{pl} - c_{pv}) / R_v} \exp{
+    \frac{L_0}{R_v T_0} - \frac{L}{R_v T}}
 
-    .. math:: 6.112 e^\frac{17.67T}{T + 243.5}
+    """
+    latent_heat = water_latent_heat_vaporization._nounit(temperature)
+    heat_power = (mpconsts.nounit.Cp_l - mpconsts.nounit.Cp_v) / mpconsts.nounit.Rv
+    exp_term = ((mpconsts.nounit.Lv / mpconsts.nounit.T0 - latent_heat / temperature)
+                / mpconsts.nounit.Rv)
+
+    return (
+        mpconsts.nounit.sat_pressure_0c
+        * (mpconsts.nounit.T0 / temperature) ** heat_power
+        * np.exp(exp_term)
+    )
```

Comment:
This looks like it matches the paper and I don't see any better ways to approach the array math, so I'd say we're good to go here.